x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/downsample/RollupShardIndexer.java

```diff
@@ -186,20 +196,48 @@ private void checkCancelled() {
                 task.getNumIndexed(),
                 task.getNumFailed()
             );
+            task.setRollupShardIndexerStatus(RollupShardIndexerStatus.CANCELLED);
             throw new TaskCancelledException(format("Shard %s rollup cancelled", indexShard.shardId()));
         }
     }
 
     private BulkProcessor2 createBulkProcessor() {
         final BulkProcessor2.Listener listener = new BulkProcessor2.Listener() {
+            private long bulkStartTime;
```

Comment:
I was looking at `BulkProcessor2#execute(...)` method and I realise that beforeBulk and afterBulk can be called from different threads. 

I then also realised that multiple bulk requests can be executed concurrency and that we need to use the `executionId` here. Each `executionId ` should have its own `bulkStartTime`.

Reply:
This is something I wanted to avoid...in that case we need to store a map of executionId -> bulkStartTime and later on retrieve the right one in the after bulk...